api/metric/api.go

```diff
@@ -18,32 +18,51 @@ import (
 	"context"
 
 	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/metric/aggregation"
 	"go.opentelemetry.io/api/registry"
 	"go.opentelemetry.io/api/unit"
 )
 
-type MetricType int
+type Type int
 
+//go:generate stringer -type=Type
 const (
-	Invalid    MetricType = iota
-	Gauge                 // Supports Set()
-	Cumulative            // Supports Inc()
+	Invalid    Type = iota
+	Gauge           // Supports Set()
+	Cumulative      // Supports Inc(): only positive values
+	Additive        // Supports Add(): positive or negative
+	Measure         // Supports Record()
 )
 
 type Meter interface {
-	// TODO more Metric types
-	GetFloat64Gauge(ctx context.Context, gauge *Float64GaugeHandle, labels ...core.KeyValue) Float64Gauge
+	GetFloat64Gauge(context.Context, *Float64GaugeHandle, ...core.KeyValue) Float64Gauge
+	GetFloat64Cumulative(context.Context, *Float64CumulativeHandle, ...core.KeyValue) Float64Cumulative
+	GetFloat64Additive(context.Context, *Float64AdditiveHandle, ...core.KeyValue) Float64Additive
+	GetFloat64Measure(context.Context, *Float64MeasureHandle, ...core.KeyValue) Float64Measure
 }
 
 type Float64Gauge interface {
 	Set(ctx context.Context, value float64, labels ...core.KeyValue)
 }
 
+type Float64Cumulative interface {
+	Inc(ctx context.Context, value float64, labels ...core.KeyValue)
+}
+
+type Float64Additive interface {
+	Add(ctx context.Context, value float64, labels ...core.KeyValue)
+}
+
+type Float64Measure interface {
+	Record(ctx context.Context, value float64, labels ...core.KeyValue)
+}
+
 type Handle struct {
```

Comment:
What is a Handle? why does this exist?

Reply:
A handle is a common base type for any of the metrics. In this proposal, the only API difference between the metric types is the verb they use, the interpretation is left to the SDK. So the `Handle` is a thing that describes the metric independently of its verb or metric type. The properties of a metric are: variable (name), type (gauge, additive, cumulative, or measure), keys (recommended), and aggregation (recommended).